proposal/js-deprecations.d.ts

```diff
@@ -109,35 +117,54 @@ declare module '../spec/js-api' {
     /**
      * A set of deprecations to treat as fatal.
      *
-     * If a version is provided, then all deprecations released in that
-     * version or earlier will be treated as fatal. If a string is provided, it
-     * will resolve to the deprecation with that ID, or error if none exists.
+     * If a deprecation warning of any provided type is encountered during
+     * compilation, the compiler must error instead.
+     *
+     * The compiler should convert any string passed here to a `Deprecation`
+     * by indexing `Deprecations`.
+     *
+     * If a version is passed here, it should be treated equivalently to passing
+     * all active deprecations whose `deprecatedIn` version is less than or
+     * equal to it.
      *
-     * This will throw an error if a future deprecation is included here, unless
-     * that future deprecation is also passed to `futureDeprecations`. This
-     * will emit a warning if an obsolete deprecation is included here. */
-    fatalDeprecations?: (Deprecation | string | Version)[];
+     * The compiler must error if a future deprecation is included here, unless
+     * that future deprecation is also passed to `futureDeprecations`. It must
+     * emit a warning if an obsolete deprecation is included here.
+     *
+     * If a deprecation is passed both here and to `silenceDeprecations`, a
+     * warning must be emitted, but making it fatal must take precedence.
+     */
+    fatalDeprecations?: (DeprecationOrId | Version)[];
 
     /**
      * A set of active deprecations to ignore.
      *
-     * If a string is provided, it will resolve to the deprecation with that ID,
-     * or error if none exists.
+     * If a deprecation warning of any provided type is encountered during
+     * compilation, the compiler must ignore it.
      *
-     * This will throw an error if a future, obsolete, or user-authored
-     * deprecation is included here.
+     * The compiler should convert any string passed here to a `Deprecation`
+     * by indexing `Deprecations`.
+     *
+     * The compiler must error if an obsolete or user-authored is included here.
```

Comment:
This sentence is missing its object. Also, might be worth being explicit that "user-authored" means specifically `deprecations['user-authored']`.

Reply:
Done.